src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1667,6 +1830,829 @@ class Simplex_tree {
+    }
+    return nullptr;
+  }
+
+//basic methods implemented for Nodes, and not Simplex_handle. The hooks in
+//nodes_by_label_ gives access to Nodes.
+public:
+  //set of methods taking Node as input. For internal use only.
+  /** Returns the Siblings containing a simplex.*/
+  static Siblings* self_siblings(Node& node, Vertex_handle v) {
+    if (node.children()->parent() == v) {
+      return node.children()->oncles();
+    }
+    else {return node.children();}
+  }
+
+  int dimension(Node & node, Vertex_handle u) {
+    Siblings * curr_sib = self_siblings(node, u);
+    int dim = 0;
+    while (curr_sib != nullptr) {
+      ++dim;
+      curr_sib = curr_sib->oncles();
+    }
+    return dim - 1;
+  }
+/* \brief Returns true if the node in the simplex tree pointed by
+ * sh has children. node must have label u*/
+  bool has_children(Node & node, Vertex_handle u) const {
+    return (node.children()->parent() == u);
+  }
+
+private:
+  // update all extra data structures in the Simplex_tree. Must be called after all
+  //simplex insertions.
+  void update_simplex_tree_after_node_insertion(Simplex_handle sh) {
+    if constexpr(Options::link_nodes_by_label) {
+      nodes_by_label_.insert(sh);
+    }
+    //every cell is critical by default
+    if constexpr(Options::store_morse_matching) {//not necessary
+      make_critical(sh);//make Morse critical by default
+    }
+  }
+
+  // update all extra data structures in the Simplex_tree. Must be called before 
+  //all simplex removals
+  void update_simplex_tree_before_node_removal(Simplex_handle sh) {
+    if constexpr(Options::link_nodes_by_label) {
+      sh->second.unlink_hooks();//remove from lists of same label Nodes
+    }
+    if constexpr(Options::store_morse_matching) {
+      if(!critical(sh)) {//if paired, make the simplices critical
+        auto p_sh = paired_with(sh);  make_critical(p_sh); make_critical(sh);
+      }
+    }
+  }
+
+public:
+/**
+  * Add a vertex or an edge in a flag complex, as well as all
+  * simplices of its star, defined to maintain the property
+  * of the complex to be a flag complex, truncated at dimension dim_max. 
+  *
+  * In term of edges in the graph, inserting edge u,v only affects N^+(u).
+  *
+  * For a new node with label v, we first do a local expansion for
+  * computing the
+  * children of this new node, and then a standard expansion for its children.
+  * Nodes with label v (and their subtrees) already in the tree
+  * do not get affected.
+  *
+  * Nodes with label u get affected only if a Node with label v is in their same
+  * siblings set.
+  * We then try to insert "ponctually" v all over the subtree rooted
+  * at Node(u). Each
+  * insertion of a Node with v label induces a local expansion at this
+  * Node (as explained above)
+  * and a sequence of "ponctual" insertion of Node(v) in the subtree
+  * rooted at sibling nodes of the new node, on its left.
+  *
+  * @param[in] u,v              Vertex_handle representing the new edge
```

Comment:
```suggestion
  * @param[in] u,v              Vertex_handle representing the new edge
  * @param[in] fil              blabla
  * @param[in] dim_max          blabla
```
Doxygen complains about missing `fil` and `dim_max` parameters.